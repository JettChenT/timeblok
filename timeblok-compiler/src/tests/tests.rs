use crate::{compile_deterministic, ir::ExactDateTime};
use insta::{assert_snapshot, glob};
use std::fs;

macro_rules! assert_compile_snapshot {
    ($input:expr, $base_time:expr) => {
        assert_snapshot!(compile_deterministic($input, $base_time.clone()).unwrap());
    };

    ($input:expr) => {
        assert_compile_snapshot!($input, ExactDateTime::from_ymd_hms(2023, 1, 1, 0, 0, 0));
    };
}


#[cfg(not(target_family = "wasm"))]
#[test]
fn run_tests(){
    glob!("bloks/*.tb", |path| {
        let input = fs::read_to_string(path).unwrap();
        assert_compile_snapshot!(&input);
    });
}

#[cfg(target_family = "wasm")]
#[test]
fn run_tests(){
    glob!("wsm-bloks/*.tb", |path| {
        let input = fs::read_to_string(path).unwrap();
        assert_compile_snapshot!(&input);
    });
}