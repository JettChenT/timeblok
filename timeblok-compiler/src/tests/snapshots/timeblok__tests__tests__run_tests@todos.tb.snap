---
source: timeblok-compiler/src/tests/tests.rs
expression: "compile_deterministic(&input,\n        ExactDateTime::from_ymd_hms(2023, 1, 1, 0, 0, 0).clone()).unwrap()"
input_file: timeblok-compiler/src/tests/bloks/todos.tb
---
BEGIN:VCALENDAR
VERSION:2.0
PRODID:ICALENDAR-RS
CALSCALE:GREGORIAN
BEGIN:VTODO
DTSTAMP:20221231T160000Z
STATUS:NEEDS-ACTION
SUMMARY:I wanna do this
UID:0
END:VTODO
BEGIN:VTODO
DTSTAMP:20221231T160000Z
STATUS:NEEDS-ACTION
SUMMARY:I wanna do that
UID:1
END:VTODO
END:VCALENDAR

